frb_example/pure_dart/dart/lib/main.dart

```diff
@@ -160,6 +160,73 @@ void main(List<String> args) async {
+    expect(api.handleSyncTypedU32(input: Uint32List.fromList([10, 20, 30])), Uint32List.fromList([10, 20, 30]));
+  });
+  test('dart call handle_sync_typed_i64', () async {
+    expect(api.handleSyncTypedI64(input: Int64List.fromList([10, 20, 30])), Int64List.fromList([10, 20, 30]));
+  }, skip: skipWeb('Not supported by dart2js'));
+  test('dart call handle_sync_typed_u64', () async {
+    expect(api.handleSyncTypedU64(input: Uint64List.fromList([10, 20, 30])), Uint64List.fromList([10, 20, 30]));
+  }, skip: skipWeb('Not supported by dart2js'));
+  test('dart call handle_sync_typed_f32', () async {
+    expect(api.handleSyncTypedF32(input: Float32List.fromList([10.1, 20.2, 30.3])),
+        Float32List.fromList([10.1, 20.2, 30.3]));
+  });
+  test('dart call handle_sync_typed_f64', () async {
+    expect(api.handleSyncTypedF64(input: Float64List.fromList([10.1, 20.2, 30.3])),
+        Float64List.fromList([10.1, 20.2, 30.3]));
+  });
+  test('dart call handle_sync_typed_i8_zero_copy', () async {
+    expect(api.handleSyncTypedI8ZeroCopy(input: Int8List.fromList([10, 20, 30])), Int8List.fromList([10, 20, 30]));
+  });
+  test('dart call handle_sync_typed_u8_zero_copy', () async {
+    expect(api.handleSyncTypedU8ZeroCopy(input: Uint8List.fromList([10, 20, 30])), Uint8List.fromList([10, 20, 30]));
+  });
+  test('dart call handle_sync_typed_i16_zero_copy', () async {
+    expect(api.handleSyncTypedI16ZeroCopy(input: Int16List.fromList([10, 20, 30])), Int16List.fromList([10, 20, 30]));
+  });
+  test('dart call handle_sync_typed_u16_zero_copy', () async {
+    expect(api.handleSyncTypedU16ZeroCopy(input: Uint16List.fromList([10, 20, 30])), Uint16List.fromList([10, 20, 30]));
+  });
+  test('dart call handle_sync_typed_i32_zero_copy', () async {
+    expect(api.handleSyncTypedI32ZeroCopy(input: Int32List.fromList([10, 20, 30])), Int32List.fromList([10, 20, 30]));
+  });
+  test('dart call handle_sync_typed_u32_zero_copy', () async {
+    expect(api.handleSyncTypedU32ZeroCopy(input: Uint32List.fromList([10, 20, 30])), Uint32List.fromList([10, 20, 30]));
+  });
+  test('dart call handle_sync_typed_i64_zero_copy', () async {
+    expect(api.handleSyncTypedI64ZeroCopy(input: Int64List.fromList([10, 20, 30])), Int64List.fromList([10, 20, 30]));
+  }, skip: skipWeb('Not supported by dart2js'));
+  test('dart call handle_sync_typed_u64_zero_copy', () async {
+    expect(api.handleSyncTypedU64ZeroCopy(input: Uint64List.fromList([10, 20, 30])), Uint64List.fromList([10, 20, 30]));
+  }, skip: skipWeb('Not supported by dart2js'));
+  test('dart call handle_sync_typed_f32_zero_copy', () async {
+    expect(api.handleSyncTypedF32ZeroCopy(input: Float32List.fromList([10.1, 20.2, 30.3])),
+        Float32List.fromList([10.1, 20.2, 30.3]));
+  });
+  test('dart call handle_sync_typed_f64_zero_copy', () async {
+    expect(api.handleSyncTypedF64ZeroCopy(input: Float64List.fromList([10.1, 20.2, 30.3])),
+        Float64List.fromList([10.1, 20.2, 30.3]));
+  });
```

Comment:
For example, structs, enums, nested structs, ...

Reply:
So I removed the tests I added and copied each existing async test into a sync one, side by side, to indicate maintainers that we should systematically test both.

This involves a lot of copy pasta but I don't feel bad about it in unit testing, especially that this shows both features work the same but independently. Does this look fine to you too @fzyzcjy ?

Reply:
Sure, unit tests can be "WET" instead of "DRY" when we want to make it more clear

(you can mark this conversation "resolved" after seeing the comment ;) )